JT/07_bedugging_and_run_debug_configuration/run_debug_configurations/Run_Debug_configurations.md

```diff
@@ -0,0 +1,154 @@
+
+All run/debug configurations are based on templates, which implement the startup logic, define the list of parameters and their default values. The list of available templates is predefined in the installation and can only be extended via plugins. However, you can edit default parameter values in each template to streamline the setup of new run/debug configurations.
+
+### Configure the default values for a template
+
+1. Go to Run | Edit Configurations.
+2. In the left-hand pane of the run/debug configuration dialog, click Edit configuration templates….
+![ij_Run_Debug_create.png](images%2Fij_Run_Debug_create.png)
+3. In the Run/Debug Configuration Templates dialog that opens, select a configuration type.
+![ij_Run_DebugDefaults.png](images%2Fij_Run_DebugDefaults.png)
+4. Specify the desired default parameters and click OK to save the template.
+
+## Run/debug configuration folders
+***
+
+When there are many run/debug configurations of the same type, you can group them in folders, so they become easier to distinguish visually.
+
+Once grouped, the run/debug configurations appear in the list under the corresponding folders.
+![rc_folders.png](images%2Frc_folders.png)
+
+### Create a folder for run/debug configurations
+1. Go to Run | Edit Configurations.
+2. In the Run/Debug Configurations dialog, select a configuration type and click the New Folder icon on the toolbar. A new empty folder for the selected type is created.
+![rc_create_folder.png](images%2Frc_create_folder.png)
+3. Specify the folder name in the text field to the right or accept the default name.
+4. Select the desired run/debug configurations and move them under the target folder.
+5. Apply the changes. If a folder is empty, it will not be saved.
+
+**When you no longer need a folder, you can delete it. The run/debug configurations grouped under this folder will be moved under the root of the corresponding run/debug configuration type.**
+
+## Passing arguments in configuration
+
+To pass arguments to a program, you should type a list of them in the format you would use on the command line. The whitespaces will be parsed as delimiters, separating one argument from another. These arguments will be passed to the entry point of the program as an array.
+
+It is important to remember that all the command-line arguments are passed as strings, so if they are supposed to be treated as, for example, integers, you should convert them explicitly inside the program.
+
+### Arguments rules
+
+Use spaces to separate individual arguments, for example, --data text.txt.
+
+If an argument includes spaces, enclose the spaces or the argument that contains spaces in double quotes, for example, some" "arg or "some arg".
+
+If an argument includes double quotes (as part of the argument), escape the double quotes using backslashes, for example, \"quoted_value\".
+
+
+## Resources
+***
+
+- [Run/debug configurations](https://www.jetbrains.com/help/idea/run-debug-configuration.html#config-folders)
```

Comment:
Let's add 1 more resource that incldues remote debugging
https://www.jetbrains.com/help/idea/tutorial-remote-debug.html#create-run-configurations

Reply:
done